From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Thu, 31 Jul 2014 10:36:42 +0200
Subject: [245/249] staging: rtl8723au: Remove two never set variables
Origin: https://git.kernel.org/linus/a48dd468efbb75a47447f82c849b1d43255a6792

struct registry_priv.usbss_enable and
struct pwrctrl_priv.bHWPwrPindetect are never set.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/core/rtw_cmd.c        | 13 -------------
 drivers/staging/rtl8723au/core/rtw_pwrctrl.c    | 22 ----------------------
 drivers/staging/rtl8723au/include/drv_types.h   |  1 -
 drivers/staging/rtl8723au/include/rtw_pwrctrl.h |  1 -
 4 files changed, 37 deletions(-)

diff --git a/drivers/staging/rtl8723au/core/rtw_cmd.c b/drivers/staging/rtl8723au/core/rtw_cmd.c
index 25d65b27..7241a5a 100644
--- a/drivers/staging/rtl8723au/core/rtw_cmd.c
+++ b/drivers/staging/rtl8723au/core/rtw_cmd.c
@@ -203,22 +203,9 @@ void rtw_free_evt_priv23a(struct evt_priv *pevtpriv)
 
 static int rtw_cmd_filter(struct cmd_priv *pcmdpriv, struct cmd_obj *cmd_obj)
 {
-	struct drvextra_cmd_parm *pdrvextra_cmd_parm;
 	/* set to true to allow enqueuing cmd when hw_init_completed is false */
 	u8 bAllow = false;
 
-	/* To decide allow or not */
-	if (pcmdpriv->padapter->pwrctrlpriv.bHWPwrPindetect &&
-	    !pcmdpriv->padapter->registrypriv.usbss_enable) {
-		if (cmd_obj->cmdcode == GEN_CMD_CODE(_Set_Drv_Extra)) {
-			pdrvextra_cmd_parm =
-				(struct drvextra_cmd_parm *)cmd_obj->parmbuf;
-			if (pdrvextra_cmd_parm->ec_id ==
-			    POWER_SAVING_CTRL_WK_CID)
-				bAllow = true;
-		}
-	}
-
 	if (cmd_obj->cmdcode == GEN_CMD_CODE(_SetChannelPlan))
 		bAllow = true;
 
diff --git a/drivers/staging/rtl8723au/core/rtw_pwrctrl.c b/drivers/staging/rtl8723au/core/rtw_pwrctrl.c
index 7fbbdaf..d5fa26a 100644
--- a/drivers/staging/rtl8723au/core/rtw_pwrctrl.c
+++ b/drivers/staging/rtl8723au/core/rtw_pwrctrl.c
@@ -150,34 +150,12 @@ void rtw_ps_processor23a(struct rtw_adapter*padapter)
 {
 	struct pwrctrl_priv *pwrpriv = &padapter->pwrctrlpriv;
 	struct mlme_priv *pmlmepriv = &padapter->mlmepriv;
-	enum rt_rf_power_state rfpwrstate;
 
 	pwrpriv->ps_processing = true;
 
 	if (pwrpriv->bips_processing == true)
 		goto exit;
 
-	if (padapter->pwrctrlpriv.bHWPwrPindetect) {
-		rfpwrstate = RfOnOffDetect23a(padapter);
-		DBG_8723A("@@@@- #2  %s ==> rfstate:%s\n", __func__,
-			  (rfpwrstate == rf_on)?"rf_on":"rf_off");
-
-		if (rfpwrstate!= pwrpriv->rf_pwrstate) {
-			if (rfpwrstate == rf_off) {
-				pwrpriv->change_rfpwrstate = rf_off;
-				padapter->bCardDisableWOHSM = true;
-				rtw_hw_suspend23a(padapter);
-			} else {
-				pwrpriv->change_rfpwrstate = rf_on;
-				rtw_hw_resume23a(padapter);
-			}
-			DBG_8723A("current rf_pwrstate(%s)\n",
-				  (pwrpriv->rf_pwrstate == rf_off) ?
-				  "rf_off":"rf_on");
-		}
-		pwrpriv->pwr_state_check_cnts ++;
-	}
-
 	if (pwrpriv->ips_mode_req == IPS_NONE)
 		goto exit;
 
diff --git a/drivers/staging/rtl8723au/include/drv_types.h b/drivers/staging/rtl8723au/include/drv_types.h
index df55e5b..9870f87 100644
--- a/drivers/staging/rtl8723au/include/drv_types.h
+++ b/drivers/staging/rtl8723au/include/drv_types.h
@@ -111,7 +111,6 @@ struct registry_priv {
 	u8	antdiv_cfg;
 	u8	antdiv_type;
 
-	u8	usbss_enable;/* 0:disable,1:enable */
 	u8	hwpdn_mode;/* 0:disable,1:enable,2:decide by EFUSE config */
 	u8	hwpwrp_detect;/* 0:disable,1:enable */
 
diff --git a/drivers/staging/rtl8723au/include/rtw_pwrctrl.h b/drivers/staging/rtl8723au/include/rtw_pwrctrl.h
index 1fa712f..be465c2 100644
--- a/drivers/staging/rtl8723au/include/rtw_pwrctrl.h
+++ b/drivers/staging/rtl8723au/include/rtw_pwrctrl.h
@@ -199,7 +199,6 @@ struct pwrctrl_priv {
 	enum rt_rf_power_state	change_rfpwrstate;
 
 	u8	bHWPowerdown;/* if support hw power down */
-	u8	bHWPwrPindetect;
 	u8	bkeepfwalive;
 	unsigned long PS_BBRegBackup[PSBBREG_TOTALCNT];
 };
